yjit/src/codegen.rs

```diff
@@ -245,6 +245,16 @@ pub enum JCCKinds {
     JCC_JNA,
 }
 
+fn gen_increment_count(asm: &mut Assembler, counter: Counter) {
```

Comment:
I think this should check `if (get_option!(gen_stats))` like `gen_counter_incr` does. `gen_counter_incr` could share this implementation if it calls `gen_increment_count(asm, Counter::$counter_name)`.

Reply:
Done. Should we just replace gen_counter_incr! with a non-macro version?

Reply:
We could replace gen_counter_incr with the method version and use the same name as before.

Reply:
@maximecb Do you want me to do this as part of this PR? Or separately?

Reply:
Would be nice if it was in this PR.

Reply:
@maximecb Done